pkg/util/ip_reservation.go

```diff
@@ -110,11 +111,21 @@ func (ipAllocator *IPAllocator) GetUnreservedIPRange(cidr string, cloudInstances
 
 	for cidrIPRange := cloneIP(ip.Mask(ipnet.Mask)); ipnet.Contains(cidrIPRange) && err == nil; cidrIPRange, err = incrementIP(cidrIPRange, incrementStep29IPRange) {
 		overLap := false
-		ipRange := fmt.Sprint(cidrIPRange.String(), reservationRange)
 		for reservedIPRange := range reservedIPRanges {
+			_, reservedIPNet, err := net.ParseCIDR(reservedIPRange)
+			if err != nil {
+				return "", err
+			}
+			// Creating IPnet object using IP and mask
+			cidrIPNet := &net.IPNet{
+				IP:   cidrIPRange,
+				Mask: ipRangeMask,
+			}
+
 			// Find if the current IP range in the CIDR overlaps with any of the reserved IP ranges. If not, this can be returned
-			overLap, err = isOverlap(ipRange, reservedIPRange)
-			// Error while processing cidr
+			overLap, err = isOverlap(cidrIPNet, reservedIPNet)
+
+			// Error while processing ipnet
 			if err != nil {
 				return "", fmt.Errorf("Error while parsing cidr to determine overlap between %s and %s", cidrIPRange.String(), reservedIPRange)
```

Comment:
Print out the error that was returned instead?  This error message is assuming implementation details of the called function.

Reply:
Also can you go through all the errors and start the message with a lowercase letter? This is the style that we have been following in Kubernetes.